src/Compilers/Core/MSBuildTask/GenerateMSBuildAnalyzerConfig.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Linq;
+using System.Text;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.CodeAnalysis.BuildTasks
+{
+    public class GenerateMSBuildAnalyzerConfig : Task
+    {
+        [Output]
+        public string ConfigFileContents { get; set; }
+
+        [Required]
+        public ITaskItem[] MetadataItems { get; set; }
+
+        [Required]
+        public ITaskItem[] PropertyItems { get; set; }
+
+        public GenerateMSBuildAnalyzerConfig()
+        {
+            ConfigFileContents = string.Empty;
+            MetadataItems = new ITaskItem[0];
+            PropertyItems = new ITaskItem[0];
+        }
+
+        public override bool Execute()
+        {
+            StringBuilder builder = new StringBuilder();
+
+            // we always generate global configs
+            builder.AppendLine("is_global = true");
+
+            // collect the properties into a global section
+            foreach (var prop in PropertyItems)
+            {
+                builder.Append("msbuild_property.")
+                       .Append(prop.ItemSpec)
+                       .Append(" = ")
+                       .AppendLine(prop.GetMetadata("Value")); //TODO: do we need to encode/escape this?
+            }
+
+            // group the metadata items by their full path
+            var groupedItems = MetadataItems.GroupBy(i => i.GetMetadata("FullPath"));
```

Comment:
Is it possible for `GetMetadata("FullPath")` to return `null`?

Reply:
My understanding is that it's 'well known metadata item' ([https://docs.microsoft.com/en-us/visualstudio/msbuild/msbuild-well-known-item-metadata?view=vs-2019](https://docs.microsoft.com/en-us/visualstudio/msbuild/msbuild-well-known-item-metadata?view=vs-2019)) so will always have a value. @rainersigwald to verify that though. 

There is a test `ItemIsNotFullyQualifiedPath` that codifies what happens when there isn't a real item on disk. 

---
In reply to: [419752014](https://github.com/dotnet/roslyn/pull/43617#discussion_r419752014) [](ancestors = 419752014)

Reply:
Yes, follows the "you asked for meaningless garbage? I'll give you `string.Empty`" philosophy of MSBuild.

(It's not a _good_ philosophy. But it's the one we have.)